src/main/java/org/opensearch/neuralsearch/processor/AgenticQueryTranslatorProcessor.java

```diff
@@ -134,11 +162,21 @@ private void executeAgentAsync(
 
                 requestListener.onResponse(request);
             } catch (IOException e) {
-                log.error("Failed to parse agent response", e);
+                log.error(
+                    "Agentic search failed - Parse error - Agent ID: [{}], Query: [{}], Error: [{}]",
+                    agentId,
+                    agenticQuery.getQueryText(),
+                    e.getMessage()
+                );
                 requestListener.onFailure(e);
             }
         }, e -> {
-            log.error("Failed to execute agent", e);
+            log.error(
+                "Agentic search failed - Agent execution error - Agent ID: [{}], Query: [{}], Error: [{}]",
+                agentId,
+                agenticQuery.getQueryText(),
+                e.getMessage()
+            );
             requestListener.onFailure(new RuntimeException("Failed to execute agentic search", e));
```

Comment:
Can we rephrase onFailure exeception to error message present in log.error? It will remove the need of putting log.error seperately.